### PR TITLE
1.0.1/context menu agenda

### PR DESCRIPTION
**Pull request by juandgaines**
- Context menu to create task or reminder.
- first appraoch agenda repository.
- Logout flow

### CODE REVIEW

File: agenda/domain/src/main/java/com/juandgaines/agenda/domain/agenda/AgendaRepository.kt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package com.juandgaines.agenda.domain.agenda
+
+import com.juandgaines.core.domain.util.DataError
+import com.juandgaines.core.domain.util.Result
+
+interface AgendaRepository {
+    suspend fun logout() :Result<Unit, DataError.Network>
```
Comment (Kai-Cilliers) [inline agenda/domain/src/main/java/com/juandgaines/agenda/domain/agenda/AgendaRepository.kt:7]: In my opinion, the `AgendaRepository` should not contain logic for logout. I know Philipp had the same approach in his Essentials Course, but for me, I would rather move the logic out of this Repo and replace with something like this:

```
interface AuthCoreService {
    suspend fun logout(): EmptyResult<DataError.Network>
}
```

```
class AgendaOverviewViewModel(
    private val agendaRepository: AgendaRepository,
    private val authCoreService: AuthCoreService,
) : ViewModel() {

}
```

Again, I am not saying this approach is wrong, but rather questioning if this is the best place for it and would you expect a `logout` function inside a class called `AgendaRepository`?

File: agenda/presentation/src/main/java/com/juandgaines/agenda/presentation/AgendaScreen.kt [hunk 2]
```diff
@@ -46,6 +85,18 @@ fun AgendaScreen(
...
+    var pressOffSet by remember {
+        mutableStateOf(DpOffset.Zero)
+    }
+    var pressOffSetProfile by remember {
+        mutableStateOf(DpOffset.Zero)
+    }
+    val density = LocalDensity.current
+    var itemHeight by remember {
+        mutableStateOf(0.dp)
+    }
```
Comment (Kai-Cilliers) [inline agenda/presentation/src/main/java/com/juandgaines/agenda/presentation/AgendaScreen.kt:98]: It appears that you are using these states to position the dropdown menus on your UI. The `DropdownMenu`  internally uses the position of its parent layout to position itself on the screen. A common way to “anchor” it is by wrapping it in a `Box` with whatever content you want (like the profile icon).

```
Box(modifier = modifier) {
  DropdownMenu(
    expanded = isExpanded,
    onDismissRequest = onDismiss,
    modifier = Modifier
        .background(TaskyWhite)
  ) {
    menuItems.forEachIndexed { index, item ->
      // render list items
    }
  }
  content()
}
```

File: agenda/data/src/main/java/com/juandgaines/agenda/data/agenda/DefaultAgendaRepository.kt [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    private val reminderRepository: ReminderRepository,
+    private val taskRepository: TaskRepository
+) : AgendaRepository {
+    override suspend fun logout(): Result<Unit, Network> {
+        val response = safeCall {
+            tokenApi.logout()
+        }
+        if (response is Result.Success){
+            sessionManager.set(null)
+        }
```
[the remark this answers is not in the document]
Comment (juandgaines) [reply agenda/data/src/main/java/com/juandgaines/agenda/data/agenda/DefaultAgendaRepository.kt:26]: got it

File: agenda/domain/src/main/java/com/juandgaines/agenda/domain/agenda/AgendaItems.kt [hunk 4]
```diff
@@ -0,0 +1,8 @@
+package com.juandgaines.agenda.domain.agenda
+
+sealed interface AgendaItems{
+    data object Task: AgendaItems
+    data object Reminder: AgendaItems
+}
+
+val agendaItems  = listOf(AgendaItems.Task, AgendaItems.Reminder)
```
[the remark this answers is not in the document]
Comment (juandgaines) [reply agenda/domain/src/main/java/com/juandgaines/agenda/domain/agenda/AgendaItems.kt:8]: thanks, its done

File: agenda/domain/src/main/java/com/juandgaines/agenda/domain/agenda/AgendaRepository.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (juandgaines) [reply agenda/domain/src/main/java/com/juandgaines/agenda/domain/agenda/AgendaRepository.kt:7]: done

Comment (juandgaines) [reply agenda/domain/src/main/java/com/juandgaines/agenda/domain/agenda/AgendaRepository.kt:7]: you are right. I will move this

File: agenda/presentation/src/main/java/com/juandgaines/agenda/presentation/AgendaScreen.kt (hunk 2 above)
Comment (juandgaines) [reply agenda/presentation/src/main/java/com/juandgaines/agenda/presentation/AgendaScreen.kt:98]: Nice trick. I will give it a try

**Pull request merged**